src/migrations/tasks/20200126113825_cwu_init.ts

```diff
@@ -0,0 +1,163 @@
+import { makeDomainLogger } from 'back-end/lib/logger';
+import { console as consoleAdapter } from 'back-end/lib/logger/adapters';
+import Knex from 'knex';
+import { CWUOpportunityStatus } from 'shared/lib/resources/code-with-us';
+import { ProposalStatus } from 'shared/lib/resources/proposal/code-with-us';
+
+const logger = makeDomainLogger(consoleAdapter, 'migrations');
+
+export async function up(connection: Knex): Promise<void> {
+  // Create CWUOpportunity table
+  await connection.schema.createTable('cwuOpportunities', table => {
+    table.uuid('id').primary().unique().notNullable();
+    table.timestamp('createdAt').notNullable();
+    table.timestamp('updatedAt').notNullable();
+    table.uuid('createdBy').references('id').inTable('users');
+    table.uuid('updatedBy').references('id').inTable('users');
+  });
+  logger.info('Created cwuOpportunities table.');
+
+  // Create CWUOpportunityVersion table
+  await connection.schema.createTable('cwuOpportunityVersions', table => {
+    table.uuid('id').primary().unique().notNullable();
+    table.timestamp('createdAt').notNullable();
+    table.uuid('createdBy').references('id').inTable('users').notNullable();
+    table.uuid('opportunity').references('id').inTable('cwuOpportunities');
+    table.string('title');
+    table.string('teaser');
+    table.boolean('remoteOk').defaultTo(false);
+    table.string('remoteDesc');
+    table.string('location');
+    table.integer('reward');
+    table.specificType('skills', 'text ARRAY');
+    table.string('description');
+    table.timestamp('proposalDeadline');
+    table.timestamp('assignmentDate');
+    table.timestamp('startDate');
+    table.timestamp('completionDate');
+    table.string('submissionInfo');
+    table.string('acceptanceCriteria');
+    table.string('evaluationCriteria');
+  });
+  logger.info('Created cwuOpportunityVersions table.');
+
+  // Create CWUOpportunityStatus table
+  await connection.schema.createTable('cwuOpportunityStatuses', table => {
+    table.uuid('id').primary().unique().notNullable();
+    table.timestamp('createdAt').notNullable();
+    table.uuid('createdBy').references('id').inTable('users').notNullable();
+    table.uuid('opportunity').references('id').inTable('cwuOpportunities');
+    table.enu('status', Object.values(CWUOpportunityStatus)).notNullable();
+    table.string('note');
+  });
+  logger.info('Created cwuOpportunityStatuses table.');
+
+  // Create CWUOpportunityAttachment table
+  await connection.schema.createTable('cwuOpportunityAttachments', table => {
+    table.uuid('id').primary().unique().notNullable();
+    table.timestamp('createdAt').notNullable();
+    table.uuid('createdBy').references('id').inTable('users').notNullable();
```

Comment:
We actually don't need these, as the `files` table provides `createdAt` and `createdBy`